tests/webTables.spec.ts

```diff
@@ -0,0 +1,177 @@
+import { test, expect } from "@playwright/test";
+
+test.describe('OWNERS page', () => {
+    test.beforeEach(async ({ page }) => {
+        await page.goto("/");
+        await page.getByText(" Owners").click();
+        await page.locator(".dropdown-menu").getByText(" Search").click();
+    });
+
+    test('Validate the pet name city of the owner', async ({ page }) => {
+        const ownerRow = page.getByRole('row', { name: '6085555387' })
+        await expect(ownerRow.locator('td').nth(2)).toHaveText('Monona')
+        await expect(ownerRow.locator('td').last()).toHaveText('Lucky')
+    })
+
+    test('Validate owners count of the Madison city', async ({ page }) => {
+        await page.waitForSelector('tbody')
+        const allCitizensRows = page.locator('tbody tr', { has: page.locator('td').nth(2) })
+        let allMadisonCitizen = 0
+
+        for (const citizen of await allCitizensRows.all()) {
+            const cellValue = await citizen.locator('td').nth(2).textContent()
+            if (cellValue == 'Madison') {
+                allMadisonCitizen += 1
+            }
+        }
+        expect(allMadisonCitizen).toEqual(4)
+    })
+
+    test('Validate search by Last Name', async ({ page }) => {
+        const inputLastName = page.locator('#lastName')
+        await inputLastName.fill('Black')
+        const buttonFindOwner = page.getByRole('button', { name: 'Find Owner' })
+        await buttonFindOwner.click()
+
+        await page.waitForTimeout(2000)
+
+        const ownersListLoop = async (searchValue) => {
+            if (searchValue == 'Playwright') {
+                expect(await page.locator('.xd-container div').last().textContent()).toContain(`No owners with LastName starting with "${searchValue}"`)
+                return;
+            }
+
+            const ownersList = page.locator('tbody tr', { has: page.locator('td').nth(2) })
+            for (const row of await ownersList.all()) {
+                const cellValue = await row.locator('td').first().textContent()
+                expect(cellValue).toContain(searchValue)
+
+            }
+        }
+
+        ownersListLoop('Black')
+
+        await inputLastName.clear()
+        await inputLastName.fill('Davis')
+        await buttonFindOwner.click()
+        await page.waitForTimeout(2000)
+        ownersListLoop('Davis')
+
+        await inputLastName.clear()
+        await inputLastName.fill('Es')
+        await buttonFindOwner.click()
+        await page.waitForTimeout(2000)
+        ownersListLoop('Es')
+
+        await inputLastName.clear()
+        await inputLastName.fill('Playwright')
+        await buttonFindOwner.click()
+        await page.waitForTimeout(2000)
```

Comment:
Remove all hardcoded waits like this from the tests. Replace it with a dynamic wait

Reply:
Got it